docs/shrc.md

```diff
@@ -66,43 +140,138 @@ The [-Xshareclasses](xshareclasses.md) option is highly configurable, allowing y
+    Setting this option allows your application to start even if there is a problem opening or creating the shared cache. In this situation, the VM might be able to start *without* class data sharing.
+
+When the cache is active, you can adjust the soft maximum size by using the [`-Xshareclasses:adjustsoftmx`](xshareclasses.md#adjustsoftmx) cache utility.
+
+Using the [`-Xshareclasses:listAllCaches`](xshareclasses.md#listallcaches) cache utility allows you to identify all the caches that exist on a system, including compatible and incompatible caches.
+
+### Creating layer caches
+
+Creating a layered cache might be useful when you are building a Docker image.
+
+Normally, writing to an existing shared cache in a lower image layer results in Docker duplicating the shared cache to the top layer (following the Docker [copy-on-write strategy](https://docs.docker.com/storage/storagedriver/#the-copy-on-write-cow-strategy)). With a layered cache, you can instead write into a new cache in the top layer. The new cache builds on the existing cache, so space is saved in the image.
 
+The following example shows a Docker container with four layers:
+
+: ![This diagram is explained in the surrounding text](./cr/shrc_layers.jpg "Docker container with layered caches")
+
+
+1. The lowest layer is a Ubuntu Docker image.
+2. The next layer is an OpenJ9 Docker image that is built on the Ubuntu image. As part of this image, the `-Xshareclasses:name=Cache1` suboption is used to create a cache called `Cache1`. The layer number assigned to this cache is `0`. The `listAllCaches` suboption shows the cache and the layer number:
+
+        ```
+        java -Xshareclasses:listAllCaches
+        ...
+        Cache name              level         cache-type      feature         layer       OS shmid       OS semid       last detach time
+
+        Compatible shared caches
+        Cache1                  Java8 64-bit  persistent      cr              0                                         Mon Sep 23 11:41:04 2019
+        ```
+
+3. The next Docker layer up is an Open Liberty image that is built on the OpenJ9 image. As part of this image, the `-Xshareclasses:name=Cache1,layer=1` suboption is used to create another cache called `Cache1`. Because the `layer=1` suboption is specified, this new cache is a layered cache, which builds on `Cache1` in the previous container layer. (Open Liberty starts two VMs, so if you instead use the `createLayer` suboption here, two layered caches are created, with layer numbers of 1 and 2.) Note that cache layers are different from, and independent of, container layers.
+
+4. In the same way, another Docker layer is added for an Open Liberty Java application, and another layered cache is created to add to `Cache1`. The `listAllCaches` suboption now shows all the caches and their layers:
```

Comment:
"Open Liberty" doesn't need to be mentioned here, "application" is sufficient.

Reply:
I altered (4). I also went to alter (3), which has 2 mentions, but am unsure whether you intended this or not. Please confirm:

*3. The next Docker layer up is an **application** image that is built on the OpenJ9 image. As part of this image, the `-Xshareclasses:name=MyCache,layer=1` suboption is used to create another cache called `MyCache`. Because the `layer=1` suboption is specified, this new cache is a layered cache, which builds on `MyCache` in the previous container layer. (**For example**, Open Liberty starts two VMs, so if you instead use the `createLayer` suboption here, two layered caches are created, with layer numbers of 1 and 2.) Note that cache layers are different from, and independent of, container layers.*

Reply:
"Open Liberty starts two VMs" is a valid warning, I think it should stay. Otherwise we can be generic in point 4.

Reply:
If the .jpg can be changed, it also mentions Open Liberty.

Reply:
We can put middleware at cache layer 1 in the picture.